### PR TITLE
usersubjects and usertasks

### PR DESCRIPTION
**Pull request by naid**

### CODE REVIEW

File: app/Models/UserSubject.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+
+class UserSubject extends Model
+{
+    const STATUS_START = 0;
+    const STATUS_TRAINING = 0;
+    const STATUS_FINISH = 0;
```
Comment (wataridori) [inline app/Models/UserSubject.php]: Why are all `status` `0`?

Comment (naid) [inline app/Models/UserSubject.php]: This is now fixed and the PR also updated.

File: app/Models/UserSubject.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        return $this->belongsTo(User::class, 'user_id');
+    }
+    public function subject()
+    {
+        return $this->belongsTo(Subject::class, 'subject_id');
+    }
+
+    public function tasks()
+    {
+        return $this->hasMany(Task::class, 'user_id', 'subject_id', 'course_id');
```
Comment (wataridori) [inline app/Models/UserSubject.php]: Why do you pass 3 parameters into `hasMany` function?

Comment (naid) [inline app/Models/UserSubject.php]: This will is now also fixed, the whole table is a pivot table so the whole relationship is not needed

**Comment by wataridori** [comment]
OK

**Comment by dinhhoanglong91** [comment]
ok

**Pull request merged**